### PR TITLE
Add uncertainty chart in the over-view section

### PR DESCRIPTION
**Pull request by priyesh777**
## Addresses:
- Issue: https://github.com/collective-service/ifrc-gates-dashboard/issues/74#issue-1342540069
- Issue: https://github.com/collective-service/ifrc-gates-dashboard/issues/33#issue-1326688233

## Changes:
- Add area chart that show the uncertainty range of given data
- Add a modal that will be used in the map hover 

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

File: app/components/PercentageStats/index.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ function PercentageStats(props: Props) {
         <ContainerCard
             className={_cs(className, styles.percentageStatsCard)}
             headingClassName={styles.percentageHeading}
-            heading={heading ? `Total number of ${heading} cases` : 'Total number of cases'}
+            heading={heading ? `${heading}` : 'Total number of cases'}
```
Comment (shreeyash07) [inline app/components/PercentageStats/index.tsx]: Revert it back

Comment (priyesh777) [reply app/components/PercentageStats/index.tsx]: Done

**Review by shreeyash07 — APPROVED** [review]

**Review by shreeyash07 — APPROVED** [review]

File: app/views/Dashboard/Overview/MapView/MapModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                        132M
+                    </Heading>
+                    <Heading
+                        className={styles.countrySurveyDate}
+                    >
+                        Nov, 2022
+                    </Heading>
+                </div>
+            )}
+            footer={<ModalFooterElement />}
```
Comment (AdityaKhatri) [inline app/views/Dashboard/Overview/MapView/MapModal/index.tsx]: Why do this?
Please put it directly here

File: app/views/Dashboard/Overview/index.tsx [hunk 3]
```diff
@@ -27,10 +31,31 @@ function Overview(props: Props) {
...
     const [currentTab, setCurrentTab] = useState<string | undefined>('mapMode');
 
+    const [
+        mapModalShown,
+        setMapModalVisible,
+        setMapModalHidden,
+    ] = useModalState(false);
+
+    const handleMapModal = () => {
+        console.log('Handle map modal:::>>>>>', mapModalShown);
```
Comment (AdityaKhatri) [inline app/views/Dashboard/Overview/index.tsx]: remove this. Also, we can directly use setMapModalVisible in the function.
Also, rename setMapModalVisible to showMapModal

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Review by shreeyash07 — APPROVED** [review]

File: app/utils/hooks/stateManagement.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import { useState, useCallback } from 'react';
+
+export function useModalState(initialValue: boolean): [
```
Comment (subinasr) [inline app/utils/hooks/stateManagement.ts]: Why not import 'useModalState' directly from 'deep-ui' ?

**Review by subinasr — CHANGES_REQUESTED** [review]

File: app/utils/hooks/stateManagement.ts (hunk 4 above)
Comment (priyesh777) [reply app/utils/hooks/stateManagement.ts]: This has been replaced by the useModalState of deep-ui

**Review by shreeyash07 — APPROVED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**